src/TraceEvent/TraceLog.cs

```diff
@@ -1786,6 +1788,95 @@ private unsafe void SetupCallbacks(TraceEventDispatcher rawEvents)
                     sampleProfileInterval100ns = data.OldInterval;
                 }
             };
+
+            var processMetadataParser = rawEvents.ProcessMetadata;
+
+            processMetadataParser.ProcessStart += delegate(ProcessStartArgsTraceData data)
+            {
+                Processes.GetOrCreateProcess((int)data.ProcessId, data.TimeStampQPC, isProcessStartEvent: true).ProcessStart(data, data.CommandLine, data.Executable, (int)data.ParentProcessId);
+            };
+
+            processMetadataParser.ProcessStop += delegate (ProcessExitArgsTraceData data)
+            {
+                Processes.GetOrCreateProcess((int)data.ProcessId, data.TimeStampQPC).ProcessEnd(data, data.CommandLine, data.Executable, (int)data.ParentProcessId, data.ExitCode);
+            };
+
+            processMetadataParser.ThreadStart += delegate (ThreadCreateArgsTraceData data)
+            {
+                TraceProcess process = processes.GetOrCreateProcess((int)data.ProcessId, data.TimeStampQPC);
```

Comment:
Can we work these more complicated Threading and module load events into helper functions and then just call them from here and from the ETW equivalents?  There's enough here that it would be nice to only have one version of these.